source/loader/ze_loader.cpp

```diff
@@ -261,16 +267,45 @@ namespace loader
         auto discoveredDrivers = discoverEnabledDrivers();
         std::string loadLibraryErrorValue;
 
-
+        const char *home_dir = "";
+#ifdef _WIN32
+        std::string home_dir_string = getenv_string("USERPROFILE");
+        if (home_dir == ""){
+            auto home_drive = getenv_string("HOMEDRIVE");
+            auto home_path = getenv_string("HOMEPATH");
+            if ((home_drive != "") && (home_path != "")) {
+                home_dir_string = home_drive + home_path;
+            } else {
+                home_dir_string = ".";
+            }
+        }
+        home_dir = home_dir_string.c_str();
+#else
+        home_dir = getenv("HOME");
+
+        if ((home_dir = getenv("HOME")) == NULL) {
```

Comment:
Why are you storing it as a C string if it needs to be a std::string at the end?

Reply:
will change